tests/integration/targets/dns_auth_nsg/tasks/main.yml

```diff
@@ -0,0 +1,215 @@
+#        filters:
+#          name: "{{ auth_nsg_name }}"
+#      register: auth_nsg_info
+#    - assert:
+#        that:
+#          - auth_nsg_info is not failed
+#          - auth_nsg_info.objects | length == 1
+#          - auth_nsg_info.objects[0].id == auth_nsg.id
+#          - auth_nsg_info.objects[0].comment == "Test Comment"
+#
+#    - name: Create Auth NSG with external primary
+#      infoblox.bloxone.dns_auth_nsg:
+#        name: "{{ auth_nsg_name }}"
+#        external_primaries:
+#          - address: "1.1.1.1"
+#            fqdn: "a.com."
+#            type: "primary"
+#        state: present
+#      register: auth_nsg
+#    - name: Get Information about the Auth NSG
+#      infoblox.bloxone.dns_auth_nsg_info:
+#        filters:
+#          name: "{{ auth_nsg_name }}"
+#      register: auth_nsg_info
+#    - assert:
+#        that:
+#          - auth_nsg is not failed
+#          - auth_nsg_info.objects | length == 1
+#          - auth_nsg_info.objects[0].external_primaries | length == 1
+#          - auth_nsg_info.objects[0].external_primaries[0].address == "1.1.1.1"
+#          - auth_nsg_info.objects[0].external_primaries[0].fqdn == "a.com."
+#          - auth_nsg_info.objects[0].external_primaries[0].type == "primary"
+#
+#    - name: Create an Auth NSG with tags
+#      infoblox.bloxone.dns_auth_nsg:
+#        name: "{{ auth_nsg_name }}"
+#        comment: "Test Comment"
+#        tags:
+#          location: "site-1"
+#        state: present
+#      register: auth_nsg
+#    - name: Get Information about the Auth NSG
+#      infoblox.bloxone.dns_auth_nsg_info:
+#        filters:
+#          name: "{{ auth_nsg_name }}"
+#      register: auth_nsg_info
+#    - assert:
+#        that:
+#          - auth_nsg_info is not failed
+#          - auth_nsg_info.objects | length == 1
+#          - auth_nsg_info.objects[0].id == auth_nsg.id
+#          - auth_nsg_info.objects[0].tags.location == "site-1"
+
+    - name: Create the Primary Auth NSG
+      infoblox.bloxone.dns_auth_nsg:
+        name: "{{ auth_nsg_name }}"
+        state: present
+      register: auth_nsg
+    - name: Create the Secondary Auth NSG referencing Primary
+      infoblox.bloxone.dns_auth_nsg:
+        name: "{{ secondary_auth_nsg_name }}"
+        nsgs:
+          - "{{ auth_nsg.id }}"
+        state: present
+      register: secondary_auth_nsg
+    - name: Get Information about the Secondary Auth NSG
+      infoblox.bloxone.dns_auth_nsg_info:
+        filters:
+          name: "{{ secondary_auth_nsg_name }}"
+      register: secondary_auth_nsg_info
+    - name: Validate that the Secondary Auth NSG references the Primary
+      assert:
+        that:
+          - secondary_auth_nsg_info is not failed
+          - secondary_auth_nsg_info.objects | length == 1
+          - secondary_auth_nsg_info.objects[0].nsgs | length == 1
+          - secondary_auth_nsg_info.objects[0].nsgs[0] == auth_nsg.id
+    - block: # Unlinking block
```

Comment:
The Unlinking is quite unnecessary . If the the unlinking is performed because it's interfering with the deletion of the primary object , why not delete the secondary object first in the `always` section and then cleanup the Primary.